packages/compass-logging/src/react.ts

```diff
@@ -0,0 +1,45 @@
+import createLoggerAndTelemetry from './logger';
+import type { LoggerAndTelemetry } from './logger';
+
+export function useLoggerAndTelemetry(
+  component: string,
+  React: { useRef: any }
+): LoggerAndTelemetry {
+  const loggerRef = React.useRef();
+  if (!loggerRef.current) {
+    loggerRef.current = createLoggerAndTelemetry(component);
+  }
+  return loggerRef.current as LoggerAndTelemetry;
+}
+
+export function useTrackOnChange(
+  component: string,
+  value: unknown,
+  onChange: (track: LoggerAndTelemetry['track']) => void,
+  options: { skipOnMount: boolean } = { skipOnMount: false },
+  React: { useRef: any; useEffect: any }
+) {
+  const onChangeRef = React.useRef(onChange);
+  onChangeRef.current = onChange;
+  const { track } = useLoggerAndTelemetry(component, React);
+  let initial = true;
+  React.useEffect(() => {
+    if (options.skipOnMount && initial) {
+      initial = false;
+      return;
+    }
+    onChangeRef.current(track);
+  }, [value, track]);
+}
+
+export function withLoggerAndTelemetry<T = any>(
+  ReactComponent: T,
+  component: string,
+  React: any
+): T {
+  const WithLoggerAndTelemetry = (props: any) => {
```

Comment:
Do we want to add typing here so that `logger` is handled as an explicit prop, similar to what we do in `compass-preferences-model`?

Reply:
Makes sense, yeah, I think in this case would make sense to remove this prop from the component interface, similar to what redux connect does for the properties mapped from state. Will update!